catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/test/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/WfsSourceTest.java

```diff
@@ -368,33 +367,18 @@ private void setUpMocks(
       }
     }
 
-    // GetFeature Response
-    when(mockWfs.getFeature(any(GetFeatureType.class))).thenReturn(mockFeatureCollection);
-
-    when(mockFeatureCollection.getFeatureMembers())
-        .thenAnswer(
-            new Answer<List<Metacard>>() {
-
-              @Override
-              public List<Metacard> answer(InvocationOnMock invocation) {
-                // Create as many metacards as there are features
-                Integer resultsToReturn = numResults;
-                if (resultsToReturn == null && numFeatures != null) {
-                  resultsToReturn = numFeatures;
-                }
-                if (resultsToReturn != null) {
-                  List<Metacard> metacards = new ArrayList<>(resultsToReturn);
-                  for (int i = 0; i < resultsToReturn; i++) {
-                    MetacardImpl mc = new MetacardImpl();
-                    mc.setId("ID_" + (i + 1));
-                    metacards.add(mc);
-                  }
-
-                  return metacards;
-                }
-                return Collections.emptyList();
-              }
-            });
+    final int resultsToReturn =
+        numResults != null ? numResults : (numFeatures != null ? numFeatures : 0);
```

Comment:
✏️ I follow the logic here, but nested ternaries can be hard to follow. Consider breaking out this assignment to a quick method instead

Reply:
I ended up refactoring the tests so this conditional logic is no longer needed. Hopefully the tests are easier to understand now.